fix: Add documentation to address AWS deployment issues (#151)
serverpod/serverpod_docs: docs/07-deployments/04-deploying-to-aws.md

By default, the Terraform scripts are configured to use a minimal setup to reduce costs for running your Serverpod. You can quickly turn on additional features, such as enabling Redis or adding a staging server by changing values in the script. You can also change these values later and redo the deployment step.

Finally, to complete your Serverpod configuration, you will need to edit the `config/staging.yaml` and `config/production.yaml` files. In these files you should:

1. replace the `examplepod.com` domain with the domain you are using for your server.

> 2. replace the database with the database url from the RDS. Replace the url with the corresponding environment yaml file in `mypod_server/config` under the `database.host` section

Reviewer: Nit: Missing a . at the end

Reviewer: Also capitalize the first letter.

```bash
aws rds describe-db-instances --db-instance-identifier ${YOUR_DB_INSTANCE_ID} | jq ".DBInstances.[0].Endpoint.Address" -r

> ```

Reviewer: Suggestion: Add a description on what this does, I'm assuming it fetches the database URL, but spell that out in text!

:::info

> ensure that you have ssl enabled for the corresponding environment as RDS enable ssl by default

Reviewer: Nit: 

```suggestion
Ensure that you have ssl enabled for the corresponding environment as RDS enable ssl by default.
```

Reviewer: Perhaps also add a reference to the yaml config file in this text.

## Deploy your code

:::caution

> using an old version of serverpod cli will generate the github action file containing old dart version. You can make the following fixes:

Reviewer: Nit:

```suggestion
Using an old version of Serverpod cli will generate the GitHub action file containing old dart version. You can make the following fixes:
```

> 1. in `./github/workflows/deployment-aws.yml` use

Reviewer: Nit:

```suggestion
1. In `./github/workflows/deployment-aws.yml` use:
```

```yaml
      - name: Setup Dart SDK
        uses: dart-lang/setup-dart@v1.6.5
        with:

>           sdk: 3.5

Reviewer: Suggetsion: Add a comment here saying that you should use the dart version you want to run your project in

e.g.
```
# Set your desired dart version here
```

```

> 1. in `mypod_server/config/production.yaml` change the version of installed dart like example

Reviewer: Question: This looks wrong, there are no dart versions defined in production.yaml should be some other file right?

Also capitalize the first letter and add a . at the end.

```bash
wget -q https://storage.googleapis.com/dart-archive/channels/stable/release/3.5.1/sdk/dartsdk-linux-x64-release.zip

> ```

Reviewer: Suggestion: Add a description on how to find the latest dart version or the list of all these URLs. Perhaps how to modify the version. That is not completely obvious right now.

## Deploy your code

:::caution

Using an old version of Serverpod cli will generate the Github action file containing old dart version that might not be the one you are using.
You can fix this by the following example. In the example, we are using the dart version v3.5.1. You can adjust to the dart version that you are using.

1. In `./github/workflows/deployment-aws.yml` use

```yaml
      - name: Setup Dart SDK
        uses: dart-lang/setup-dart@v1.6.5
        with:
          sdk: 3.5
```

> 1. In `server/deploy/aws/terraform/init-script.sh` change the version of installed dart like example

Reviewer: Nit:

```suggestion
1. In `server/deploy/aws/terraform/init-script.sh` change the version of dart installed on the machine.
```

You can find a list of your currently running EC2 instances by navigating to _EC2 > Instances_. Click on one of the instances to go to its summary page. From there, click _Connect_. On the _Connect to instance_ page, click _Connect_, and AWS will open up a console window with access to your EC2 instance.

After Signing in to your instance, you should check if the service is running with `systemctl status serverpod.service`

> If the service is running, you can look into the serverpod error log in `serverpod.err` and server log in `serverpod.log` in the home directory

Reviewer: Nit: Add a . at the end.

> If all checks out, try to use `curl localhost:8082` to see if the service can be reached from local. If we get the expected response, we know that the service is running properly.

Reviewer: Suggestion: change the port to the API port as that is guaranteed to be running, the web server can potentially be turned off depending on how you have configured Serverpod.

e.g.

curl localhost:8080
